app/code/Magento/ProductAlert/Controller/Unsubscribe/Stock.php

```diff
@@ -21,16 +23,33 @@ class Stock extends UnsubscribeController
     protected $productRepository;
 
     /**
-     * @param \Magento\Framework\App\Action\Context $context
-     * @param \Magento\Customer\Model\Session $customerSession
-     * @param \Magento\Catalog\Api\ProductRepositoryInterface $productRepository
+     * @var Magento\Store\Model\StoreManagerInterface
+     */
+    protected $storeManager;
+
+    /**
+     * @var \Magento\ProductAlert\Model\StockFactory
+     */
+    protected $stockFactory;
+
+    /**
+     * Stock constructor.
+     * @param Context $context
+     * @param CustomerSession $customerSession
+     * @param ProductRepositoryInterface $productRepository
+     * @param StoreManagerInterface $storeManager
+     * @param \Magento\ProductAlert\Model\StockFactory $stockFactory
      */
     public function __construct(
         Context $context,
         CustomerSession $customerSession,
-        ProductRepositoryInterface $productRepository
+        ProductRepositoryInterface $productRepository,
+        StoreManagerInterface $storeManager,
```

Comment:
Please add default value `null` to keep this changes Backward compatible.

Reply:
@swnsma
done